File: output/ChatGPT/avatar/Python/C++/codeforces_583_B.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }
    int current_level = 0;
    int counter = 0;
    int counter_turns = -1;
    while (counter < a.size()) {
        counter_turns++;
        for (int i = 0; i < a.size(); i++) {
            if (a[i] != -1 && a[i] <= counter) {
                counter++;
                a[i] = -1;
            }
        }
        reverse(a.begin(), a.end());
    }
    cout << counter_turns << endl;

    return 0;
}

File: output/CodeGen/avatar/Python/Java_source/codeforces_583_B.java
public class codeforces_583_B {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = in.nextInt();
        }
        int currentLevel = 0;
        int counter = 0;
        int counterTurns = -1;
        while (counter < n) {
            counterTurns += 1;
            for (int i = 0; i < n; i++) {
                if (a[i]!= -1 && a[i] <= counter) {
                    counter += 1;
                    a[i] = -1;
                }
            }
            a.reverse();
        }
        System.out.println(counterTurns);
    }
}